### PR TITLE
[Kernel] Update the Javastyle to enforce brace style and fix instances

### PR DESCRIPTION
**Pull request by allisonport-db**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [x] Kernel
- [ ] Other (fill in here)

## Description

Updates the java checkstyle for Kernel to enforce
- Left { braces are at the end of the line https://checkstyle.org/checks/blocks/leftcurly.html#LeftCurly
- Right } braces are on the same line as the next block start https://checkstyle.org/checks/blocks/rightcurly.html#RightCurly
- Fixes the import order

## How was this patch tested?

Checkstyle passes

### CODE REVIEW

File: kernel/dev/checkstyle.xml [hunk 1]
```diff
@@ -198,7 +201,8 @@
         <module name="ImportOrder">
             <property name="separated" value="true"/>
             <property name="ordered" value="true"/>
-            <property name="groups" value="java.,javax.,scala,*,io.delta.kernel,io.delta.kernel.internal"/>
```
Comment (vkorukanti) [inline kernel/dev/checkstyle.xml:201]: Given the `default` namespace going to be `io.delta.kernel.default.internal` for default internal, do you want to add it to the order?

**Review by vkorukanti — APPROVED** [review]
lgtm, one minor comment.

File: kernel/dev/checkstyle.xml (hunk 1 above)
Comment (allisonport-db) [reply kernel/dev/checkstyle.xml:201]: once we make that change how do we want our imports ordered in the default module? do we want the "default" module packages grouped separately?

<java, scala>

<3rd party>

<io.delta.kernel>
<io.delta.kernel.internal>

<io.delta.kernel.default>
<io.delta.kernel.default.internal>

Comment (vkorukanti) [reply kernel/dev/checkstyle.xml:201]: No strong preference, my main question was to not mix the `io.delta.kernel.default.internal` together with `io.delta.kernel`.

Also like the above order.

**Pull request closed without merging**